lib/charset.c

```diff
@@ -443,10 +478,13 @@ static void qp2byte(struct convert_rock *rock, uint32_t c)
 static void b64_2byte(struct convert_rock *rock, uint32_t c)
 {
     struct b64_state *s = (struct b64_state *)rock->state;
-    char b = CHAR64(c);
+    char b = CHAR64(c, s->index);
 
-    /* could just be whitespace, ignore it */
-    if (b == XX) return;
+    if (b >= XS) {
```

Comment:
This is changing the base64 parsing behaviour to treat everything other than specific whitespace as `invalid`, which is probably fine but it does mean that all our base64 parsing is going to abort on stuff that it previously accepted.  I would want to make this a 'syslog but allow' at least for now, for non-URL base64.